CONTRIBUTING.md

```diff
@@ -3,28 +3,76 @@
-1. Create a pull request on that branch for the development team to talk about and discuss the suggested change.  The PR triggers Continuous Integration tests which will initially fail.
-1. Do any development necessary in core or hdk crates of this repo to actually implement the feature demonstrated in `app_spec`
-1. Finally, when the feature is fully implemented, the CI tests should turn green and the branch can be merged.
+The simplest way to report a bug is via [Github Issues](https://github.com/holochain/holochain/issues/new/choose) by selecting the Bug Report issue type. Please fill out all relevant areas of the bug report, including steps to reproduce. If your report is not specific enough, we will have a hard time addressing it without further followup.
 
-In this way `app_spec` works as a living specification with example app to build against. -->
+## Bug fixes and minimal reproductions
+
+We gladly welcome pull requests that help us identify and fix bugs!
+
+The end goal of addressing any bug is to have a test written in our codebase to reproduce the bug, and of course to implement the fix for the bug. A PR with at least a minimal reproduction demonstrating the bug is extremely helpful, even if the fix has not been discovered.
+
+To write a minimal reproduction of a problem discovered "in the wild", we recommend you to write a sample zome, DNA, or hApp which demonstrates the problem, and open a PR with your failing test. We have a library called [`sweettest`](https://docs.rs/holochain/latest/holochain/sweettest/index.html) which is well-suited to the task of testing the behavior of Holochain applications. 
+
+When writing your reproduction PR, you can recreate the problematic part of your app in one of two ways: "inline zomes", or "test wasms".
+
+### How to create "inline zomes"
+
+The quickest, most preferable way to reproduce a problem is through "inline zomes". Inline zomes are written in terms of a collection of functions, like normal Wasm zomes, but they don't get compiled to wasm, and instead are run inline by Rust directly. This lightweight approach to writing zomes is well suited for quick test cases, or for cases that require a multitude of zomes in order to reproduce a problem.
+
+To create a test based on inline zomes, see existing tests using `InlineZomeSet` or `SweetInlineZomes` for guidance. Just put your test in a place that feels appropriate.
+
+### How to create a "test wasm"
+
+Holochain has many "test wasms", which are sample zomes written to demonstrate specific functionality. These
+
+It may be necessary to write a test wasm if the problem you've encountered has to do with the actual machinery of executing Wasm code. It may also be a good option if you discovered a problem while writing a zome yourself, in which case you can simply copy and paste the offending code into a new test wasm.
```

Comment:
```suggestion
It may be necessary to write a test wasm if the problem you've encountered has to do with the actual machinery of executing wasm code. It may also be a good option if you discovered a problem while writing a zome yourself, in which case you can simply copy and paste the offending code into a new test wasm.
```

Reply:
The canonical representation that I see is "Wasm", like on [the wikipedia page](https://en.wikipedia.org/wiki/WebAssembly)

Reply:
Good, I'm fine with any spelling, be it wasm, Wasm or WASM.